packages/flutter_tools/gradle/build.gradle

```diff
@@ -0,0 +1,39 @@
+plugins {
+    id 'groovy-gradle-plugin'
+}
+
+sourceSets {
+    main {
+        groovy {
+            srcDirs += ['.']
+        }
+    }
+}
+
+repositories {
+    mavenCentral()
+    google()
+}
+
+group = 'dev.flutter'
+version = '1.0.0'
+
+gradlePlugin {
+    // Define the plugin
+    plugins {
+        blahblah {
+            id = 'dev.flutter.flutter-gradle-plugin'
+            implementationClass = 'FlutterPlugin'
+        }
+    }
+}
+
+dependencies {
+    // When bumping, also update ndkVersion in FlutterExtension, as well as the
+    // Android Gradle Plugin version in ../lib/src/android/gradle_utils.dart.
+    compileOnly("com.android.tools.build:gradle:7.2.2")
+}
+
+tasks.withType(Jar) {
+    duplicatesStrategy = DuplicatesStrategy.WARN
```

Comment:
What does this bit do?

Reply:
It was a temporary mitigation - already removed it. The problem was that the plugin's generated files were overlapping for some strange reason, and Gradle failed with the message along the lines of "duplicates detected, define duplicatesStrategy explicitly". So I did, and it was happy.

After I fixed the plugin structure to be more standard in https://github.com/flutter/flutter/pull/123511/commits/451b238d2d5b7658dc4446af028bd1aec4c03a06, the problem with duplicates went away :)